### ISSUE TITLE
Hyperdata linkValue removed on save

### ISSUE THREAD

**Issue by mariuswestgaard** [verbb/hyper#215]
### Describe the bug

I have a hyper field on a global set, where I select multiple links. Whenever I have filled the fields in global set and saved it it looks good, the entry and settings are there. When resaving the set, the entry is being removed.

The entries being selected is entries from a section with "Save entries to other sites in the same site group", and the selected entries are all properly propagated to each site in the group.
The hyper field is also "Translate for each site group".

On first save I see this data in logs:
```
'hyperData' => [
        209436697 => [
            'linkValue' => [
                0 => '170'
            ]
            'linkSiteId' => '1'
            'linkText' => 'Personbil'
        ]
    ]
```

On resave (I see the following data in the logs:
```
'hyperData' => [
        1312617210 => [
            'linkValue' => ''
            'linkSiteId' => ''
            'linkText' => 'Personbil'
        ]
    ]
```

It's the same result if I have added more than one link. It's then like this on resave:
```
'hyperData' => [
        1312617210 => [
            'linkValue' => ''
            'linkSiteId' => ''
            'linkText' => 'Personbil'
        ]
        1610208295 => [
            'linkValue' => ''
            'linkSiteId' => ''
            'linkText' => 'Varebil'
        ]
        ... and the rest of links to follow
```

Edit: It seems it's only happening when it's more than 1 links added to the field.

### Steps to reproduce

1. Add a hyper field to a global set.
2. Add multiple links with entries (that exists on multiple sites)
3. Save
4. Looks good.
5. Save again
6. The entry is unselected

### Craft CMS version

5.5.3

### Plugin version

2.2.0

### Multi-site?

Yes

**Comment by engram-design** [comment]
I can't seem to replicate that, sorry! Wondering if you could shoot through a video of this in action, just in case I'm missing something. I've setup new fields and sections just to test that, and things _seem_ to be working correctly.

**Comment by mariuswestgaard** [comment]
Sorry for the late reply. I just noticed this issue seems to be fixed by updating craft to 5.5.5. I can't replicate it here now, so it appears it was a Craft issue. Thanks anyway!
